core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -377,6 +360,41 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
         }
     }
 
+    private void abortLargeRequest(ChannelHandlerContext ctx, DecodedHttpRequestWriter decodedReq, int id,
+                                   boolean endOfStream, boolean isEarlyRejection,
+                                   KeepAliveHandler keepAliveHandler) {
+        final ContentTooLargeException cause =
+                ContentTooLargeException.builder()
+                        .maxContentLength(decodedReq.maxRequestLength())
+                        .transferred(decodedReq.transferredBytes())
+                        .contentLength(decodedReq.headers())
+                        .setEarlyRejection(isEarlyRejection)
+                        .build();
+        if (!isEarlyRejection) {
```

Comment:
i cannot mark it as discarding or null the req - I need to pass it down to HttpServerHandler I believe.

Reply:
I didn't understand this point - isn't the `fireChannelRead` called on the local variable?

https://github.com/line/armeria/blob/36bb54d3b4fc961d700bd3a42be05d69225feb26/core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java#L294

I think it's probably fine to just keep the old behavior irregardless of whether it is earlyRejection or not

i.e.
```java
discarding = true;
req = null;
```

Reply:
good point, updated